.github/eks-workflow-files/jio-eks/jio-template.yaml

```diff
@@ -0,0 +1,549 @@
+                name: ray
+                protocol: TCP
+              - containerPort: 24430
+                name: ray-client
+                protocol: TCP
+
+              resources:
+                limits:
+                  nvidia.com/gpu: "8"
+                  vpc.amazonaws.com/efa: "32"
+                requests:
+                  nvidia.com/gpu: "8"
+                  vpc.amazonaws.com/efa: "32"
+
+              securityContext:
+                capabilities:
+                  add:
+                  - IPC_LOCK
+                  - SYS_PTRACE
+                  - NET_ADMIN
+                privileged: true
+
+              volumeMounts:
+              - name: shmem
+                mountPath: /dev/shm
+
+            nodeSelector:
+              node.kubernetes.io/instance-type: p5.48xlarge
+
+            tolerations:
+            - key: nvidia.com/gpu
+              operator: Exists
+              effect: NoSchedule
+
+            volumes:
+            - name: shmem
+              emptyDir:
+                medium: Memory
+                sizeLimit: 150Gi
+
+  # JAX Node (Node 1)
+  - name: jax-node
+    replicas: 1
+    template:
+      spec:
+        backoffLimit: 0
+        completionMode: Indexed
+        completions: 1
+        parallelism: 1
+        template:
+          metadata:
+            labels:
+              node-role: jax
+          spec:
+            serviceAccountName: jax-worker
+            hostNetwork: true
+            dnsPolicy: ClusterFirstWithHostNet
+
+            affinity:
+              podAntiAffinity:
+                requiredDuringSchedulingIgnoredDuringExecution:
+                - labelSelector:
+                    matchExpressions:
+                    - key: node-role
+                      operator: In
+                      values:
+                      - vllm
+                  topologyKey: kubernetes.io/hostname
+
+            containers:
+            - name: jax-container
+              image: PLACEHOLDER # here we need JAX_VLLM_OFFLOADING_IMAGE
+              command:
+              - bash
+              - -c
+              - |
+                set -euo pipefail
+                # Define Variables
+                GATEWAY_HOST="${JOBSET_NAME}-gateway-0-0.${JOBSET_NAME}"
+                export GATEWAY_URL="${GATEWAY_HOST}:50051"
+
+                HOST_IP=$(hostname -I | awk '{print $1}')
+                export JAX_COORDINATOR_ADDRESS="${HOST_IP}:12345"
+                export JAX_COORDINATOR_PORT="12345"
+
+                echo "=== Starting on JAX Node (Node ${JOB_COMPLETION_INDEX}) ==="
+                echo "Host IP: ${HOST_IP}"
+                echo "Gateway URL: ${GATEWAY_URL}"
+                echo "JAX Coordinator: ${JAX_COORDINATOR_ADDRESS}"
+                echo "JAX Process Index: ${JAX_PROCESS_INDEX}"
+                echo "JAX Process Count: ${JAX_NUM_PROCESSES}"
+
+                # Wait for gateway
+                echo "Waiting for gateway at ${GATEWAY_HOST}:50051..."
```

Comment:
Making a note for myself to add a `timeout` argument to the bridge API when connecting to the gateway.